src/Spectre.Service/Loaders/DatasetLoader.cs

```diff
@@ -0,0 +1,136 @@
+*/
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using Ninject;
+using Spectre.Data.Datasets;
+using Spectre.Dependencies;
+using Spectre.Service.Configuration;
+
+namespace Spectre.Service.Loaders
+{
+    /// <summary>
+    /// Class for loading datasets from specified directories.
+    /// </summary>
+    public class DatasetLoader
+    {
+        #region Fields
+
+        /// <summary>
+        /// Root for local directory.
+        /// </summary>
+        private readonly string _localRoot;
+
+        /// <summary>
+        /// Root for remote directory.
+        /// </summary>
+        private readonly string _remoteRoot;
+
+        #endregion
+
+        #region Constructor
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DatasetLoader"/> class.
+        /// </summary>
+        /// <param name="dataRootConfig">Validated configuration of needed directories.</param>
+        public DatasetLoader(DataRootConfig dataRootConfig)
+        {
+            _localRoot = dataRootConfig.LocalPath;
+            _remoteRoot = dataRootConfig.RemotePath;
+
+            FileSystem = DependencyResolver.GetService(typeof(IFileSystem)) as IFileSystem;
+        }
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Handle to file system.
+        /// </summary>
+        public IFileSystem FileSystem { private get; set; }
+
+        #endregion
+
+        #region Loaders
+
+        /// <summary>
+        /// Returns dataset basing directly on the name of the file.
+        /// </summary>
+        /// <param name="name">File name.</param>
+        /// <returns>Found dataset.</returns>
+        /// <exception cref="FileNotFoundException">Throws when the file is not found both locally and remotely.</exception>
+        /// <exception cref="DatasetFormatException">Throws when the loader fails to create dataset from the file.</exception>
+        public IDataset GetFromName(string name)
+        {
+            string fullPathLocal = Path.Combine(_localRoot, name);
+
+            if (!FileSystem.File.Exists(fullPathLocal))
+            {
+                string fullPathRemote = Path.Combine(_remoteRoot, name);
+                if (!FileSystem.File.Exists(fullPathRemote))
+                {
+                    throw new DatasetNotFoundException("Dataset file not found neither locally nor remotely.", name);
+                }
+
+                FileSystem.File.Copy(fullPathRemote, fullPathLocal);
+            }
+
+            try
+            {
+                return new BasicTextDataset(fullPathLocal);
+            }
+            catch (IOException e)
+            {
+                FileSystem.File.Delete(fullPathLocal);
+                throw new DatasetFormatException($"Failed to load dataset from file '{name}'.", e);
+            }
+        }
+
+        /*
+        //  TODO: @dkuchta: complete the methods below when database access is available
```

Comment:
Add it on Trello, either as a task, or extend the database-related task.

Reply:
Done